cmd/minikube/cmd/root.go

```diff
@@ -73,6 +74,10 @@ var RootCmd = &cobra.Command{
 // Execute adds all child commands to the root command sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	_, callingCmd := filepath.Split(os.Args[0])
+	if callingCmd == "kubectl" {
+		os.Args = append([]string{"minikube", callingCmd}, os.Args[1:]...)
```

Comment:
would this work in cases in integration test if the binary is called

"minikube-darwin-amd64" ? or only works if binary is called "minikube"?  do simmilar tools, support alternative binary names such as "minkube-linux-amd64" ? or "minikube.exe"

Reply:
Interesting question. I think the binary name doesn't matter once we reach such instruction. This is what comes next:
```
for _, c := range RootCmd.Commands()
```
As long as minikube can correctly understand the content of "RootCmd.Commands()" the binary name shouldn't matter. And I don't see any logic for different platforms. Am I right?

Reply:
Instead of hardcoding "minikube" I've opted for using the value within the `RootCmd.Use` variable. I assume this will handle all the different cases.